app/controllers/galleries_controller.rb

```diff
@@ -1,21 +1,27 @@
 # frozen_string_literal: true
 class GalleriesController < ApplicationController
+  helper_method :documents
```

Comment:
This is a clever and succinct way of preventing the API calls when there is cached HTML... _but_ API calls should really be triggered earlier than view rendering.

I'd suggest that we do something like move the body_cache_key generation for each action (and corresponding view) into the controller and have it check for the existence of that cache entry before making the API call.